chat/room.go

```diff
@@ -54,18 +54,20 @@ type Room struct {
 	commands  Commands
 	closed    bool
 	closeOnce sync.Once
+	checkName func( *message.User ) error
 
 	Members *set.Set
 }
 
 // NewRoom creates a new room.
-func NewRoom() *Room {
+func NewRoom( checkName func( *message.User ) error ) *Room {
```

Comment:
Please avoid changing public function signatures when possible (for backwards compatibility and also minimizing interface concerns). In this case, we can avoid passing in `checkName` as an argument by making it a new public element on the `Room` struct that can be set after `NewRoom()` is called.